src/librbd/mirror/DemoteRequest.h

```diff
@@ -53,6 +64,10 @@ class DemoteRequest {
    */
 
   ImageCtxT &m_image_ctx;
+  const int64_t m_group_pool_id;
+  const std::string m_group_id;
+  const std::string m_group_snap_id;
```

Comment:
This class is common to both Journal and snapshot mirroring and should ideally not have snap related attributes.